### PR TITLE
show more details on connection close

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

File: broker/src/tunneldigger_broker/tunnel.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def close(self, reason=protocol.ERROR_REASON_UNDEFINED):
         :param reason: Reason code for the tunnel being closed
         """
 
-        logger.info("Closing tunnel %d (%s) after %d seconds", self.tunnel_id, self.uuid, time.time() - self.created_time)
+        logger.info("Closing tunnel {} ({}) after {} seconds (reason=0x{:x})".format(self.tunnel_id, self.uuid, int(time.time() - self.created_time), reason))
```
Comment (mitar) [inline broker/src/tunneldigger_broker/tunnel.py:246]: Nit: Don't pre-format in logging. :-)

Comment (RalfJung) [reply broker/src/tunneldigger_broker/tunnel.py:246]: I know you prefer the `%` style but hey formatting is so much easier with the modern syntax...

Comment (RalfJung) [reply broker/src/tunneldigger_broker/tunnel.py:246]: I know you prefer the % style but hex formatting is so much easier with the modern syntax...

Comment (mitar) [reply broker/src/tunneldigger_broker/tunnel.py:246]: True. :-)

You could also do:

```
logger.info("Closing tunnel %d (%s) after %d seconds (reason=%s)", self.tunnel_id, self.uuid, time.time() - self.created_time, "0x{:x}".format(reason))
```

**Comment by mitar** [comment]
BTW, "This branch is out-of-date with the base branch".

**Review by mitar — APPROVED** [review]

File: broker/src/tunneldigger_broker/tunnel.py [hunk 2]
```diff
@@ -330,6 +330,7 @@ def message(self, address, msg_type, msg_data, raw_length):
         if msg_type == protocol.CONTROL_TYPE_ERROR:
             # Error notification from the remote side.
             # TODO: Parse error code.
+            logger.warning("Tunnel %d (%s) got error from remote peer", self.tunnel_id, self.uuid)
```
Comment (mitar) [inline broker/src/tunneldigger_broker/tunnel.py:333]: What about parsing the error code and logging it?

Comment (RalfJung) [reply broker/src/tunneldigger_broker/tunnel.py:333]: I mean, that would be nice, but for now having the code printed is better than nothing I think.

**Pull request merged**